sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -474,6 +474,7 @@ class DataFrame private[sql](
           val rightCol = withPlan(joined.right).resolve(col).toAttribute.withNullability(true)
           Alias(Coalesce(Seq(leftCol, rightCol)), col)()
         }
+      case NaturalJoin(_) => sys.error("NaturalJoin with using clause is not supported.")
```

Comment:
yup - although we should still throw some exception here just in case we refactor code in the future so this is reachable.